### PR TITLE
Add buildkite CI

### PR DESCRIPTION
**Pull request by vchuravy**
- [DroneCI] Develop MPIPreferences
- remove old gitlab config
- also run CI against release- branches

### CODE REVIEW

**Comment by vchuravy** [comment]
@giordano Can drone also run Julia 1.7?

**Comment by giordano** [comment]
Drone uses the official julia docker images, which in turn rely on the official Julia binaries. We have had official binaries for aarch64 for pretty much all releases, so yes, you can use 1.7 too. I find it slightly easier to edit the (now gone) jsonnet file: you can't do a matrix in the yaml file, so you have to copy over everything for a different version of Julia

File: .drone.yml [hunk 1]
```diff
@@ -11,11 +11,13 @@ steps:
   image: julia:1.6
   commands:
   - "julia --project=. --check-bounds=yes --color=yes -e 'using InteractiveUtils; versioninfo(verbose=true)'"
+  - "julia --project=. --color=yes -e 'using Pkg; Pkg.develop(path="lib/MPIPreferences")'"
```
[the remark this answers is not in the document]
Comment (vchuravy) [reply .drone.yml:14]: Or if indeed I need the Jason I was confused by the duplication.

Can I just have the Jason thingy?

[the remark this answers is not in the document]
Comment (giordano) [reply .drone.yml:14]: Sadly no :sob:  The jsonnet is nicer, but drone only actually reads the yaml, but there is a [CLI tool](https://docs.drone.io/pipeline/scripting/jsonnet/) to convert the jsonnet to yaml.

[the remark this answers is not in the document]
Comment (vchuravy) [reply .drone.yml:14]: Oh no... I already hate it

[the remark this answers is not in the document]
Comment (giordano) [reply .drone.yml:14]: Yeah, they could just do the conversion server-side...

**Comment by vchuravy** [comment]
Hm despite doing a sync before passing the data we still see:

```
Test Failed at /var/lib/buildkite-agent/builds/gpuci-6/julialang/mpi-dot-jl/test/test_bcast.jl:25
  Expression: B == A
   Evaluated: Float32[0.29905963 0.23247325 … 0.81700486 0.6008139; 0.4981863 0.10537517 … 0.70482075 0.8041256; … ; 0.4952435 0.07156503 … 0.8349975 0.91180354; 0.38014913 0.05967331 … 0.52544254 0.11160207] == Float32[0.08002287 0.46079296 … 0.0022229552 0.53184825; 0.86832696 0.65267456 … 0.17802185 0.35201555; … ; 0.672473 0.40036875 … 0.8157358 0.19764471; 0.36474752 0.2532596 … 0.817343 0.5458178]
ERROR: LoadError: There was an error during testing
in expression starting at /var/lib/buildkite-agent/builds/gpuci-6/julialang/mpi-dot-jl/test/test_bcast.jl:21
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
```

**Pull request merged**